nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -36,8 +38,17 @@ def _apply_mask_and_get_affinity(seeds, niimg, radius, allow_overlap,
 
         X = masking._apply_mask_fmri(niimg, mask_img)
     else:
-        mask_coords = list(np.ndindex(niimg.shape[:3]))
-        X = niimg.get_data().reshape([-1, niimg.shape[3]]).T
+        #Adding  warning message 
+        if np.isnan(np.sum(_safe_get_data(niimg))):
+            warnings.warn('The imgs you have fedded into fit_transform()' 
+                          'contains NaN values which are converted to zeroes '
+                          'please use mask_imgs to mask out NaNs othewise you '
```

Comment:
'please use .....' is not needed. We can end the message at '... converted to zeros.'